src/main/java/net/neoforged/neoforge/common/MutableDataComponentHolder.java

```diff
@@ -74,16 +74,20 @@ default <T> T remove(Supplier<? extends DataComponentType<? extends T>> componen
     }
 
     /**
-     * Copes a data component from {@code src} into {@code target}
+     * Copies a data component from {@code src}
+     * 
+     * @implNote This will clear the current component value if the requested {@code src} holder does not contain a matching {@code componentType} value.
      */
-    static <T> void copyFrom(DataComponentHolder src, MutableDataComponentHolder target, DataComponentType<T> componentType) {
-        target.set(componentType, src.get(componentType));
+    default <T> void copyFrom(DataComponentHolder src, DataComponentType<T> componentType) {
```

Comment:
I have a slight preference for the first argument being the `DataComponentType`, for consistency with the other methods.